daliuge-engine/dlg/apps/simple.py

```diff
@@ -909,7 +885,10 @@ def run(self):
 
 
 class SimpleBranch(BranchAppDrop, NullBarrierApp):
-    """Simple branch app that is told the result of its condition"""
+    """
+    Simple branch app that is told the result of its condition.
+    We are keeping this not to break existing graphs.
+    """
```

Comment:
The existing graphs don't really achieve anything - is there value in keeping those (and by extension, this)? I think it would be better to remove them all, especially given we need to clean up the EAGLE-graph-repo anyway.

Reply:
That is (mostly) true. Will have a look into how much work this is.